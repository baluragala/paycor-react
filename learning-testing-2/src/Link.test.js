import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallowToJson } from "enzyme-to-json";
import Link from "./Link";

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });
describe("Link", () => {
  it("should render props correctly", () => {
    const wrapper = shallow(
      <Link url="https://www.zeolearn.com" title="zeolearn" />
    );
    expect(wrapper.props().href).toEqual("https://www.zeolearn.com");
    expect(wrapper.props().children).toEqual("zeolearn");
  });

  it("should handle state changes", () => {
    const wrapper = shallow(
      <Link url="https://www.zeolearn.com" title="zeolearn" />
    );
    expect(wrapper.state().clicked).toEqual(false);
    wrapper.simulate("click");
    expect(wrapper.state().clicked).toEqual(true);
  });

  it("should have called window.alert", () => {
    window.alert = jest.fn();
    const wrapper = shallow(
      <Link url="https://www.zeolearn.com" title="zeolearn" />
    );
    wrapper.simulate("click");
    expect(window.alert).toHaveBeenCalledWith("clicked!");
  });

  it("should have good snapshot", () => {
    const wrapper = shallow(
      <Link url="https://www.zeolearn.com" title="zeolearn" />
    );
    const snapshot = shallowToJson(wrapper);
    expect(snapshot).toMatchSnapshot();
  });
});
