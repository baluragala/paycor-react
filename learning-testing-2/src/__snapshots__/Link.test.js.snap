// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should have good snapshot 1`] = `
<a
  href="https://www.zeolearn.com"
  onClick={[Function]}
>
  zeolearn
</a>
`;
